doc/development.md

```diff
@@ -0,0 +1,127 @@
+# Development
+
+A Go 1.7+ environment is required. Docker and rkt (with SELinux set to Permissive) should be installed.
+
+## Binaries
+
+Develop the `update-operator` and `update-agent` apps locally.
+
+```
+make
+```
+
+Build  the `update-operator` and `update-agent` binaries in a build image with rkt.
+
+```
+make release-bin
+```
+
+## Container Image
+
+Build a container image.
+
+```
+make image
+```
+
+Push a development image to a personal image repository to test it on a cluster.
+
+```
+export IMAGE_REPO=quay.io/USERNAME/container-linux-update-operator
+make docker-push
+```
+
+## Verificaton
+
+### Cluster
+
+Deploy a Container Linux Kubernetes cluster that satisfies the [requirements](README.md#requirements).
+
+* [QEMU/KVM](https://github.com/coreos/matchbox/tree/master/examples/terraform/bootkube-install)
+* [Tectonic](https://github.com/coreos/tectonic-installer)
+
+In particular, be sure to mask `locksmithd.service` on every Container Linux node.
+
+```
+sudo systemctl mask locksmithd.service --now
+```
+
+### Deploy
+
+Edit `example/update-operator.yaml` `image` to refer to the development image. Edit the command to tell the operator which development image to use `- --agent-image-repo=quay.io/USERNAME/container-linux-update-operator`.
+
+Create the `update-operator` deployment.
+
+```
+kubectl create -f examples/update-operator.yaml
+```
+
+### Checks
+
+Verify the `update-operator` is able to acquire a leader lock.
+
+```sh
+$ kubectl logs container-linux-update-operator-1096583598-x1m6t -n kube-system
+I0622 18:42:13.594217       1 main.go:46] /bin/update-operator running
+I0622 18:42:13.791474       1 leaderelection.go:179] attempting to acquire leader lease...
+I0622 18:42:13.840638       1 leaderelection.go:189] successfully acquired lease kube-system/container-linux-update-operator-lock```
+```
+
+Verify a Daemonset `container-linux-update-agent-ds` was created.
+
+```sh
+$ kubectl get daemonsets -n kube-system
+```
+
+Currently, the `update-operator` has two different notions of version, tagged versions like `v0.2.0-dev` and commit versions like the GIT_SHA. `update-operator` assumes if should deploy an `update-agent` of the same tag version, whereas Docker images are tagged with Git tag or commit hash appropriately. Workaround this by adding a Docker tag to your development image `v3.0.0-dev`. The pods in the `container-linux-update-agent` Daemonset will then be able to pull the image.
```

Comment:
This can also be removed once #100 merges. We're moving away from `manage-agent=true`.